config/kernel-monitor.json

```diff
@@ -4,6 +4,7 @@
 	"lookback": "5m",
 	"bufferSize": 10,
 	"source": "kernel-monitor",
+	"metricsReporting": true,
```

Comment:
I remembered that you used a different approach previously, i.e., allow setting the metric name. Why do we change it to a global bool?

Reply:
Setting the metric name seems more flexible to me.
- It allows using arbitrary metric name
- It enables reporting only some metrics from the problem, but not all of them (e.g., only report metrics when metric name is provided).

Reply:
Sorry my previous implementation was a mistake, and deviated from my original [design doc](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit#bookmark=id.fgbb40a0mr45).

So basically we can choose how to represent problems as metrics, in one of the below two forms:

- problem_counter{"name": "KernelDeadlock"}  0
- problem_kernel_deadlock  0 

Basically, the choice is whether to put the problem type in the metric name, or metric label. Both designs offer equal level of "query-bility". i.e. Any metrics query you can do with one design, you can always translate it easily to a query on the other design.

I think we should put them in metric label by default, see the reasoning [here](https://docs.google.com/document/d/1rcPBMJChlFXbrPpXHlE-gk6DtUQh6p85zqu5Aq3ySuc/edit#bookmark=id.w57y0telmrx).
tl;dr: Mainly because we want to be able to add new problem types easily. Adding a metrics endpoint typically requires some monitoring backend side review, and can be slow. Adding a metrics label value is very lightweight.


In the [design doc](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit#bookmark=id.fgbb40a0mr45), I also mentioned this as extension:

> And we plan to support some more customized behaviors using the config file. For example:
>  - Report the metric as gauge, rather than counter.
>  - Allow user to change the metric name.
>  - ...

This PR is implementing 1).
I don't think 2) is needed right now. When we might need it, I'm happy to submit a PR for it.

Reply:
Discussed offline. Will go with problem_counter approach.